flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/MultiJoinSemanticTests.java

```diff
@@ -26,6 +28,14 @@
 /** Semantic tests for {@link StreamExecMultiJoin}. */
 public class MultiJoinSemanticTests extends SemanticTestBase {
 
+    @Override
+    protected void envDefaultOptions(TableEnvironment env) {
+        env.getConfig()
+                .set(
+                        OptimizerConfigOptions.TABLE_OPTIMIZER_NONDETERMINISTIC_UPDATE_STRATEGY,
```

Comment:
let's move this to the default implementation in the base class, all semantic tests should run with this by default.

Reply:
I think this is a good idea. However, I think we need other unrelated changes to enable it, including PTFs. Tried it quickly and 

```
java.lang.UnsupportedOperationException: Unsupported to visit node StreamPhysicalProcessTableFunction, please add the visit implementation if it is a newly added stream physical node.
```

So I'd suggest for us to do it in a next PR.

Reply:
1. https://issues.apache.org/jira/browse/FLINK-38233
2. https://issues.apache.org/jira/browse/FLINK-38234